### PR TITLE
Upgrade jsonrpsee to v0.4.1

### PR DESCRIPTION
**Pull request by dvdplm**
Upgrade `remote-externalities` to use the latest `jsonrpsee` version. 

~~q: is there a way to avoid adding the `DeserializeOwned` bound everywhere? :/~~ No, unfortunately not, as the `Hash` is part of the return type.

polkadot companion: https://github.com/paritytech/polkadot/pull/4256

### CODE REVIEW

**Comment by dvdplm** [comment]
@kianenigma PTAL?

File: utils/frame/remote-externalities/src/lib.rs [hunk 1]
```diff
@@ -40,23 +48,29 @@ pub mod rpc_api;
 type KeyPair = (StorageKey, StorageData);
 
 const LOG_TARGET: &str = "remote-ext";
-const DEFAULT_TARGET: &str = "wss://rpc.polkadot.io";
+const DEFAULT_TARGET: &str = "wss://rpc.polkadot.io:443";
```
[the remark this answers is not in the document]
Comment (kianenigma) [reply utils/frame/remote-externalities/src/lib.rs:52]: what's the difference?

**Review by kianenigma — APPROVED** [review]
Thanks!

File: utils/frame/remote-externalities/src/lib.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (niklasad1) [reply utils/frame/remote-externalities/src/lib.rs:52]: after we moved to `http::Uri` for parsing the URI/URL default port number doesn't work, https://github.com/paritytech/jsonrpsee/issues/554

**Review by emostov — APPROVED** [review]

**Comment by emostov** [comment]
Look like CI is complaining about an unused import https://gitlab.parity.io/parity/substrate/-/jobs/1212818

**Comment by dvdplm** [comment]
bot merge

**Comment by dvdplm** [comment]
bot merge

**Comment by dvdplm** [comment]
bot merge

**Pull request merged**